snap/channel/channel_test.go

```diff
@@ -248,6 +254,37 @@ func (s *storeChannelSuite) TestFull(c *C) {
 	}
 }
 
+func (s *storeChannelSuite) TestFuncFull(c *C) {
+	tests := []struct {
+		channel string
+		str     string
+	}{
+		{"stable", "latest/stable"},
+		{"latest/stable", "latest/stable"},
+		{"1.0/edge", "1.0/edge"},
+		{"1.0/beta/foo", "1.0/beta/foo"},
+		{"1.0", "1.0/stable"},
+		{"candidate/foo", "latest/candidate/foo"},
+		// store behaviour compat; expect these to fail when we stop accommodating the madness :)
+		{"//stable//", "latest/stable"},
+		// rather weird corner case
+		{"///", ""},
+		// empty string is OK
+		{"", ""},
+	}
+
+	for _, t := range tests {
+		can, err := channel.Full(t.channel)
+		c.Assert(err, IsNil)
+		c.Check(can, Equals, t.str)
+	}
+}
+
+func (s *storeChannelSuite) TestFuncFullErr(c *C) {
+	_, err := channel.Full("foo/bar/baz/quux")
```

Comment:
Do need a test here that checks if we catch invalid risk levels? I.e. "1.0/no-risk-level" and "1.0/no-risk-level/foo" ? Or is that not an error ?

Reply:
it's not an error at this level, at least with the current code :-)